drivers/serial/Kconfig.silabs_eusart

```diff
@@ -8,8 +9,23 @@ config UART_SILABS_EUSART
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
 	select SOC_GECKO_EUSART
+	select SERIAL_SUPPORT_ASYNC \
+		if DT_HAS_SILABS_LDMA_ENABLED
+	select DMA if UART_ASYNC_API
 	select PINCTRL
 	select CLOCK_CONTROL
 	select PM_DEVICE if PM
 	help
 	  Enable the eusart uart driver.
+
+if UART_SILABS_EUSART
+
+config UART_SILABS_EUSART_ASYNC
+	bool
+	depends on DMA_SILABS_LDMA
+	depends on UART_ASYNC_API
```

Comment:
It does not make sense to enable DMA if `UART_ASYNC_API` is not used?

Reply:
For code size reasons, since UART only uses DMA in the async API, we should IMO avoid compiling with DMA support if async API is disabled. UART is different from SPI and I2C, where DMA can also be used for blocking transfers, since UART as 3 separate APIs for single-character polling, interrupt-driven transfers and async transfers.